Add OSMesa headless mode, run WPT against Webrender.

glennw opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/13382)

<!-- Reviewable:end -->

gw3583 committed:
Add OSMesa headless mode, run WPT against Webrender, update tests.

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/constellation/Cargo.toml
 * @wafflespeanut: python/servo/command_base.py, python/servo/testing_commands.py, python/servo/post_build_commands.py
 * @jgraham: tests/wpt/harness/wptrunner/browsers/servodriver.py, tests/wpt/harness/wptrunner/wptcommandline.py, tests/wpt/harness/wptrunner/browsers/servo.py, tests/wpt/harness/wptrunner/executors/executorservo.py, tests/wpt/harness/requirements_servo.txt
 * @kichjang: components/script/Cargo.toml, components/script/dom/webidls/Window.webidl, components/script/lib.rs, components/net/Cargo.toml, components/script/dom/webidls/Element.webidl

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 5ad48af with merge 5abb39f...

glennw commented:
This is ready for review now.

If anyone interested can take a look at the test expectation changes and let me know if there's anything in there that we don't want to land with updated expectations, I'll look further into the root cause of the differing results.

@emilio r? on the webgl paint thread changes.

@jdm r? on the script / webidl changes.

@larsbergstrom r? on the osmesa build changes.

@emilio / @larsbergstrom  / @pcwalton r? on the glutin headless / osmesa changes.

@jgraham / @larsbergstrom r? on the python / wpt changes. Some parts (e.g. the WPT / python changes) might be in the wrong place - please let me know if the case.

bors-servo commented:
:broken_heart: Test failed - [linux-dev](http://build.servo.org/builders/linux-dev/builds/3125)

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit aa19938 with merge e31279a...

bors-servo commented:
:broken_heart: Test failed - [linux-dev](http://build.servo.org/builders/linux-dev/builds/3127)

jdm commented:
@glennw Could you verify if all the other new crashes in the test expectations are the same matrix problem?

glennw commented:
@jdm I've verified that the four panics in the transform tests are the same bug, and that the three panics in the filters tests are https://github.com/servo/webrender/issues/419.

Both of them can be fixed before landing this if you'd prefer (the filters one is a bit more complicated to fix than the transform one though).

jdm commented:
I'm fine with landing them as is. No need to hold them up as long as we understand what's going on.

jdm commented:
r=me on the script/webidl changes.

emilio commented on components/canvas/webgl_paint_thread.rs:
```diff
@@ -122,47 +114,61 @@ impl WebGLPaintThread {
-                                    height as u32,
+                chan.send(CanvasData::Pixels(pixel_data)).unwrap();
+            }
+            WebGLPaintTaskData::WebRender(ref wr_api, _, Some(image_key)) => {
```
hmm... Do we have the context bound at this point in this thread? I think we don't.

glennw replied:
Ah, yes. I guess we just need to add the WR(Some(image)) case to the init() method below, right? So that it binds the context in the same way as the non-WR readback case?

emilio replied:
Hm... Yes, though you'd need to reach the actual context, plus apply commands locally in this thread instead of sending them to WebRender.

All in all, I think the best approach is to expose the RendererKind (and probably the wrapper types for the context too) and move the fallback logic here, so that the `Readback` data contains the correct kind of context.

emilio commented on ports/glutin/window.rs:
```diff
@@ -215,7 +320,7 @@ impl Window {
     fn gl_version() -> GlRequest {
         if opts::get().use_webrender {
-            return GlRequest::Specific(Api::OpenGl, (3, 2));
+            return GlRequest::Specific(Api::OpenGl, (3, 3));
```
Is this completely necessary? (I assume so, but it's kind of unfortunate)

emilio commented on ports/glutin/window.rs:
```diff
@@ -671,44 +811,89 @@ fn create_window_proxy(_: &Window) -> Option<glutin::WindowProxy> {
+            WindowKind::Window(ref window) => {
+                window.set_inner_size(size.width as u32, size.height as u32)
+            }
+            WindowKind::Headless(..) => {}
```
I think we can simplify this in these cases returning early and adding a `window()` method, or something, but I guess this is fine. r=me on the glutin changes.

jgraham commented:
wptrunner changes look mostly fine, but need to also land upstream.

---

Reviewed 4 of 4 files at r9, 1 of 1 files at r12.
Review status: 5 of 155 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[python/servo/command_base.py, line 54 at r14](https://reviewable.io:443/reviews/servo/servo/13382#-KSLUmu4BqldsUilrClR:-KSLUmu5jNr3nuPFZoOr:b6wbt0t) ([raw file](https://github.com/servo/servo/blob/aa1993872b582b2dff80efc178ca574212a0a492/python/servo/command_base.py#L54)):*
> ```Python
>     deps_path = path.join(path.split(bin_path)[0], "build")
>     with cd(deps_path):
>         print(os.getcwd())
> ```

Intentional?

---

*[tests/wpt/harness/requirements_servo.txt, line 2 at r14](https://reviewable.io:443/reviews/servo/servo/13382#-KSLSVh_fKVRk6goZDVN:-KSLSVh_fKVRk6goZDVO:b7dnsv5) ([raw file](https://github.com/servo/servo/blob/aa1993872b582b2dff80efc178ca574212a0a492/tests/wpt/harness/requirements_servo.txt#L2)):*
> ```
> mozprocess >= 0.19
> Mako >= 1.0.4
> ```

Hmm, why is this required to run the tests?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/13382#-:-KSLVF2N1dlpZGwF1Hhc:b-761c7r)*
<!-- Sent from Reviewable.io -->

glennw commented:
Review status: 5 of 155 files reviewed at latest revision, 5 unresolved discussions.

---

*[tests/wpt/harness/requirements_servo.txt, line 2 at r14](https://reviewable.io:443/reviews/servo/servo/13382#-KSLSVh_fKVRk6goZDVN:-KSYd69P9busbB7Rs5Uk:b-x80dgi) ([raw file](https://github.com/servo/servo/blob/aa1993872b582b2dff80efc178ca574212a0a492/tests/wpt/harness/requirements_servo.txt#L2)):*
<details><summary><i>Previously, jgraham wrote…</i></summary>
> Hmm, why is this required to run the tests?
</details>
The osmesa build process requires python + mako. I'm assuming this is the wrong place to put the requirement, although it does work :) Where should it be?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/13382)*
<!-- Sent from Reviewable.io -->

glennw commented:
Review status: 5 of 155 files reviewed at latest revision, 5 unresolved discussions.

---

*[ports/glutin/window.rs, line 323 at r14](https://reviewable.io:443/reviews/servo/servo/13382#-KSJCFwcMj3epD88KcdY-r14-323:-KSYdYaKXZqJmCmJA_Dv:bspem3l) ([raw file](https://github.com/servo/servo/blob/aa1993872b582b2dff80efc178ca574212a0a492/ports/glutin/window.rs#L323)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary>
> Is this completely necessary? (I assume so, but it's kind of unfortunate)
</details>
Nope, fixed.

---

*[python/servo/command_base.py, line 54 at r14](https://reviewable.io:443/reviews/servo/servo/13382#-KSLUmu4BqldsUilrClR:-KSYdbnEZJHVdPW4LUTB:b-886gsx) ([raw file](https://github.com/servo/servo/blob/aa1993872b582b2dff80efc178ca574212a0a492/python/servo/command_base.py#L54)):*
<details><summary><i>Previously, jgraham wrote…</i></summary>
> Intentional?
</details>
Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/13382)*
<!-- Sent from Reviewable.io -->

glennw commented:
@jgraham What is the upstreaming process - is it something we do after this all lands? Or does it need to be done beforehand?

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit e99b022 with merge cd647fb...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3355)

glennw commented:
@bors-servo try

glennw commented:
@bors-servo retry

bors-servo commented:
:hourglass: Trying commit e99b022 with merge 89d0579...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3356)

jdm commented:
We can upstream after it lands; upgrading wptrunner has been more complicated than I expected (#13261).

glennw commented:
@jdm OK, sounds good. Do you know what the bors test failure above means?

glennw commented:
Does this bors failure mean something wrong with CI / build machines?

error: failed to open: /home/servo/.cargo/registry/index/github.com-1ecc6299db9ec823/.cargo-index-lock

jdm commented:
@bors-servo: retry
* weird cargo failure

bors-servo commented:
:hourglass: Trying commit e99b022 with merge 66a3191...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3357)

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 882bcf1 with merge 92250e6...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3360)

glennw commented:
@bors-servo retry

* weird cargo failure

bors-servo commented:
:hourglass: Trying commit 882bcf1 with merge 9631f22...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3361)

glennw commented:
It seems the CI system is resisting running WR today :| @larsbergstrom any idea what those cargo failures are and how to resolve them?

glennw commented:
@bors-servo retry

bors-servo commented:
:hourglass: Trying commit 882bcf1 with merge 585b36e...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3364)

jdm commented:
@bors-servo: retry
* cargo git failure

bors-servo commented:
:hourglass: Trying commit 882bcf1 with merge b667932...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3366)

glennw commented:
@bors-servo retry

* See if CI / git failure is still happening

bors-servo commented:
:hourglass: Trying commit 882bcf1 with merge 4294653...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3369)

jgraham commented:
The upstreaming process should be pretty easy, you just need to land the patches in https://github.com/w3c/wptrunner. I'm happy for you to merge them there without further review, so if you make a PR just ping @Ms2ger or I.

emilio commented:
@bors-servo: retry

bors-servo commented:
:hourglass: Trying commit 882bcf1 with merge dc9ca43...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3379)

highfive commented:
▶ FAIL [expected PASS] /css-backgrounds-3_dev/html4/background-size-025.htm
      └   → /css-backgrounds-3_dev/html4/background-size-025.htm 2ea3e9ca7e4e2240bd95bb6deb98ca225529708e
    /css-backgrounds-3_dev/html4/reference/background-size-025-ref.htm 7fc2ddd8cc52163a417c8aba45a68a099418a2e1
    Testing 2ea3e9ca7e4e2240bd95bb6deb98ca225529708e == 7fc2ddd8cc52163a417c8aba45a68a099418a2e1

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit ea74ed5 with merge 10d6af4...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3395)

highfive commented:
▶ FAIL [expected PASS] /_mozilla/css/border_black_ridge_a.html
      └   → /_mozilla/css/border_black_ridge_a.html d9d9e489bc5b4508c01b06c7ff92f19dcc2ece3c
    /_mozilla/css/border_black_groove.html d9d9e489bc5b4508c01b06c7ff92f19dcc2ece3c
    Testing d9d9e489bc5b4508c01b06c7ff92f19dcc2ece3c != d9d9e489bc5b4508c01b06c7ff92f19dcc2ece3c

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 5738459 with merge 1a66662...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3399)

highfive commented:
▶ FAIL [expected PASS] /css-backgrounds-3_dev/html4/background-size-025.htm
      └   → /css-backgrounds-3_dev/html4/background-size-025.htm 2ea3e9ca7e4e2240bd95bb6deb98ca225529708e
    /css-backgrounds-3_dev/html4/reference/background-size-025-ref.htm 7fc2ddd8cc52163a417c8aba45a68a099418a2e1
    Testing 2ea3e9ca7e4e2240bd95bb6deb98ca225529708e == 7fc2ddd8cc52163a417c8aba45a68a099418a2e1

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit da4327b with merge 4a325d3...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3337)

highfive commented:
▶ PASS [expected FAIL] /_mozilla/mozilla/webgl/tex_image_2d_mipmap.html

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 55ed27a with merge 9a6f264...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3339)

glennw commented:
@bors-servo retry

bors-servo commented:
:hourglass: Trying commit 55ed27a with merge 291ca15...

glennw commented:
@jdm @larsbergstrom @metajack OK, this appears to be passing now (apart from normal intermittent issues). I think most of the files have been r= by various reviewers. How should we proceed to getting this merged? Perhaps if I squash everything into one commit and then get each file signed off in reviewable?

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3342)

highfive commented:
▶ TIMEOUT [expected OK] /html/the-xhtml-syntax/parsing-xhtml-documents/xhtml-mathml-dtd-entity-support.htm
      │ 
      │ VMware, Inc.
      │ Gallium 0.4 on softpipe
      └ 3.3 (Core Profile) Mesa 12.0.1

metajack commented:
@glennw That sounds fine to me.

emilio commented:
r=me on the canvas, glutin and script (since they're only removing a WR hack) changes. The `Cargo.toml` changes look good to me (modulo that target inconsistency), though we might want to get `osmesa-src` published or something?

---

Reviewed 3 of 3 files at r4, 2 of 3 files at r7, 1 of 4 files at r15, 1 of 4 files at r16, 3 of 5 files at r17, 2 of 15 files at r20.
Review status: 17 of 144 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[ports/glutin/Cargo.toml, line 40 at r21](https://reviewable.io:443/reviews/servo/servo/13382#-KSfHqT-Q29aXNmJxWol:-KSfHqT08n_4v71RiURX:b9ji762) ([raw file](https://github.com/servo/servo/blob/55ed27a0dc081876a90ecd3a81d6dea81ff384bc/ports/glutin/Cargo.toml#L40)):*
> ```TOML
> gdi32-sys = "0.2"
> 
> [target.'cfg(target_os = "macos")'.dependencies]
> ```

Why `target_os = "linux"` in one and a specific target otherwise? Can we use `cfg(any(target_os = "linux", target_os = "macos"))` here?

---

*[ports/glutin/window.rs, line 118 at r21](https://reviewable.io:443/reviews/servo/servo/13382#-KSfIB0B5-wrh5e5IlEB:-KSfIB0C1SDQ5SgqoyjI:bi0jzyx) ([raw file](https://github.com/servo/servo/blob/55ed27a0dc081876a90ecd3a81d6dea81ff384bc/ports/glutin/window.rs#L118)):*
> ```Rust
>     #[cfg(any(target_os = "linux", target_os = "macos"))]
>     fn new(width: u32, height: u32) -> HeadlessContext {
>         let mut attribs = Vec::new();
> ```

This can be an array in the stack if you want, though I guess it's not too important.

---

*[ports/glutin/window.rs, line 125 at r21](https://reviewable.io:443/reviews/servo/servo/13382#-KSfIZab1O24OYm2mYQE:-KSfIZab1O24OYm2mYQF:b7i0ca5) ([raw file](https://github.com/servo/servo/blob/55ed27a0dc081876a90ecd3a81d6dea81ff384bc/ports/glutin/window.rs#L125)):*
> ```Rust
>         attribs.push(3);
>         attribs.push(osmesa_sys::OSMESA_CONTEXT_MINOR_VERSION);
>         attribs.push(3);
> ```

You might want to use the same core profile if possible (3.2 instead of 3.3).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/13382#-:-KSfJgFOKy8MLnbCKyW3:byt0ef4)*
<!-- Sent from Reviewable.io -->

larsbergstrom commented:
r=me for osmesa build changes

larsbergstrom commented:
Reviewed 1 of 3 files at r10, 1 of 4 files at r15, 1 of 4 files at r16.
Review status: 20 of 144 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/13382)*
<!-- Sent from Reviewable.io -->

larsbergstrom commented:
@glennw Indeed, it looks like everything is reviewed, modulo some comments from @emilio above. If the remaining test failures are just intermittents, I think we should land this. At the risk of being overly dramatic about dates, I'd like to offer my assistance in sherpa-ing this over the last hurdles to get it in by this Friday-ish (the end of Q3).

{insert snarky comment here about internal goals associated with arbitrary calendar deadlines}

glennw commented:
Review status: 20 of 144 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[ports/glutin/window.rs, line 118 at r21](https://reviewable.io:443/reviews/servo/servo/13382#-KSfIB0B5-wrh5e5IlEB:-KSh9iKzqUNE893IYeTC:blrsaj6) ([raw file](https://github.com/servo/servo/blob/55ed27a0dc081876a90ecd3a81d6dea81ff384bc/ports/glutin/window.rs#L118)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary>
> This can be an array in the stack if you want, though I guess it's not too important.
</details>
I'll leave as a vec in interests of landing this, but will fix up later.

---

*[ports/glutin/window.rs, line 125 at r21](https://reviewable.io:443/reviews/servo/servo/13382#-KSfIZab1O24OYm2mYQE:-KSh9vKtrdDMRGDeNxYa:b-8pble5) ([raw file](https://github.com/servo/servo/blob/55ed27a0dc081876a90ecd3a81d6dea81ff384bc/ports/glutin/window.rs#L125)):*
<details><summary><i>Previously, emilio (Emilio Cobos Álvarez) wrote…</i></summary>
> You might want to use the same core profile if possible (3.2 instead of 3.3).
</details>
I originally tried 3.2 but it only works with 3.3 unfortunately (I haven't investigated exactly why yet).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/13382)*
<!-- Sent from Reviewable.io -->

glennw commented:
@larsbergstrom @metajack squashed and rebased.

glennw commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 9fc64b0 with merge 8886146...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3221)

highfive commented:
▶ PASS [expected FAIL] /css-backgrounds-3_dev/html4/background-repeat-round-roundup.htm

glennw commented:
@bors-servo retry

 * Test if intermittent.

bors-servo commented:
:hourglass: Trying commit 9fc64b0 with merge 9c94cde...

larsbergstrom commented:
r=me when you're ready.

For any additional straggler tests that do not appear to be intermittent, please disable and open a bug. It's clearly VERY close and already what we're shipping in the nightly, and we've made you suffer through rebases, etc. for long enough at this point :-)

glennw commented:
@bors-servo try- r=larsbergstrom

Let's just try our luck landing this :)

bors-servo commented:
:pushpin: Commit 9fc64b0 has been approved by `larsbergstrom`

<!-- @bors-servo r=larsbergstrom 9fc64b0f014c65e417b8f4592ee6ea4a8b71c261 -->

glennw commented:
@bors-servo p=1000

KiChjang commented:
@bors-servo r- try- clean

KiChjang commented:
@bors-servo r=larsbergstrom force

bors-servo commented:
:pushpin: Commit 9fc64b0 has been approved by `larsbergstrom`

<!-- @bors-servo r=larsbergstrom 9fc64b0f014c65e417b8f4592ee6ea4a8b71c261 -->

KiChjang commented:
@bors-servo r=larsbergstrom

bors-servo commented:
:pushpin: Commit e9e8760 has been approved by `larsbergstrom`

<!-- @bors-servo r=larsbergstrom e9e8760ca8ba89e4557e466c6c59c53401129acb -->

bors-servo commented:
:hourglass: Testing commit e9e8760 with merge 45d4349...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3437)

highfive commented:
▶ TIMEOUT [expected OK] /XMLHttpRequest/send-entity-body-document.htm
      │ 
      │ VMware, Inc.
      │ Gallium 0.4 on softpipe
      └ 3.3 (Core Profile) Mesa 12.0.1

glennw commented:
@bors-servo r=larsbergstrom

bors-servo commented:
:pushpin: Commit cdb69d4 has been approved by `larsbergstrom`

<!-- @bors-servo r=larsbergstrom cdb69d442317e5ba57d571913007ae258879afc1 -->

bors-servo commented:
:hourglass: Testing commit cdb69d4 with merge c7e1a57...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3438)

notriddle commented:
`/css21_dev/html4/max-width-017.htm` failed, and it's not a known intermittent.

glennw commented:
@bors-servo retry

 * Testing if intermittent - have never seen that test fail before locally or during all these try runs.

bors-servo commented:
:zap: Previous build results for [arm32](http://build.servo.org/builders/arm32/builds/3177), [arm64](http://build.servo.org/builders/arm64/builds/3191), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3261), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3245), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3228), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1141) are reusable. Rebuilding only [linux-rel](http://build.servo.org/builders/linux-rel/builds/3438), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3369)...

notriddle commented:
It was a CI fire, actually.

glennw commented:
@bors-servo retry

bors-servo commented:
:zap: Previous build results for [arm32](http://build.servo.org/builders/arm32/builds/3177), [arm64](http://build.servo.org/builders/arm64/builds/3191), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3261), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3245), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3228), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1141) are reusable. Rebuilding only [linux-rel](http://build.servo.org/builders/linux-rel/builds/3438), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3369)...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3177), [arm64](http://build.servo.org/builders/arm64/builds/3191), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3261), [linux-rel](http://build.servo.org/builders/linux-rel/builds/3442), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3245), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3228), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3379), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1141)
